lib/executor.js

```diff
@@ -330,24 +332,37 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
-    data.config = getUpdatedConfigObject(app, data.config);
+    var useEnvVar = true;
+    data.config = getUpdatedConfigObject(app, data.config, useEnvVar);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
-function getUpdatedConfigObject(app, config) {
+function getUpdatedConfigObject(app, config, useEnvVar) {
```

Comment:
Boolean arguments are difficult to read at call-site and also difficult to extend. Please use an `options` argument instead.

```js
// usage
getUpdatedConfigObject(app, obj, { useEnvVar: true });

// impl
function getUpdatedConfigObject(app, config, options) {
  var useEnvVar = options && options.useEnvVar;
  // ...
}
```

Also please rename the variable/flag to `useEnvVars` (plural form), since there can be more than a single env variable to handle.